src/Microsoft.NetCore.Analyzers/Core/Security/UseDefaultDllImportSearchPathsAttribute.cs

```diff
@@ -0,0 +1,143 @@
+                    return;
+                }
+
+                var cancellationToken = compilationStartAnalysisContext.CancellationToken;
+                var unsafeDllImportSearchPathBits = compilationStartAnalysisContext.Options.GetUnsignedIntegralOptionValue(
+                    optionName: EditorConfigOptionNames.UnsafeDllImportSearchPathBits,
+                    rule: DoNotUseUnsafeDllImportSearchPathRule,
+                    defaultValue: UnsafeBits,
+                    cancellationToken: cancellationToken);
+                var defaultDllImportSearchPathsAttributeOnAssembly = compilation.Assembly.GetAttributes().FirstOrDefault(o => o.AttributeClass.Equals(defaultDllImportSearchPathsAttributeTypeSymbol));
+
+                compilationStartAnalysisContext.RegisterSymbolAction(symbolAnalysisContext =>
+                {
+                    var symbol = symbolAnalysisContext.Symbol;
+
+                    if (!symbol.IsExtern || !symbol.IsStatic)
+                    {
+                        return;
+                    }
+
+                    var dllImportAttribute = symbol.GetAttributes().FirstOrDefault(s => s.AttributeClass.Equals(dllImportAttributeTypeSymbol));
+                    var defaultDllImportSearchPathsAttribute = symbol.GetAttributes().FirstOrDefault(s => s.AttributeClass.Equals(defaultDllImportSearchPathsAttributeTypeSymbol));
+
+                    if (dllImportAttribute != null)
+                    {
+                        var constructorArguments = dllImportAttribute.ConstructorArguments;
+
+                        if (constructorArguments.Length == 0)
+                        {
+                            return;
+                        }
+
+                        if (Path.IsPathRooted(constructorArguments[0].Value.ToString()))
+                        {
+                            return;
+                        }
+
+                        var rule = UseDefaultDllImportSearchPathsAttributeRule;
+                        var ruleArgument = symbol.Name;
+
+                        if (defaultDllImportSearchPathsAttribute == null)
+                        {
+                            if (defaultDllImportSearchPathsAttributeOnAssembly != null)
+                            {
+                                var dllImportSearchPathOnAssembly = (int)defaultDllImportSearchPathsAttributeOnAssembly.ConstructorArguments.FirstOrDefault().Value;
+                                var validBits = dllImportSearchPathOnAssembly & unsafeDllImportSearchPathBits;
+
+                                if (dllImportSearchPathOnAssembly != LegacyBehavior &&
+                                    validBits == 0)
+                                {
+                                    return;
+                                }
+
+                                rule = DoNotUseUnsafeDllImportSearchPathRule;
+                                ruleArgument = ((DllImportSearchPath)validBits).ToString();
+                            }
```

Comment:
Not a huge deal, but may want to consider refactoring so you don't have to repeat this code and the code below.